### ISSUE TITLE
`<dialog bind:open>` shows error

### ISSUE THREAD

**Issue by hyunbinseo** [sveltejs/eslint-plugin-svelte#612]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.53.0

### What version of `eslint-plugin-svelte` are you using?

2.35.0

### What did you do?

<details>
<summary>Configuration</summary>

```ts
// SvelteKit default
module.exports = {
  root: true,
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:svelte/recommended',
    'prettier',
  ],
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  parserOptions: {
    sourceType: 'module',
    ecmaVersion: 2020,
    extraFileExtensions: ['.svelte'],
  },
  env: {
    browser: true,
    es2017: true,
    node: true,
  },
  overrides: [
    {
      files: ['*.svelte'],
      parser: 'svelte-eslint-parser',
      parserOptions: {
        parser: '@typescript-eslint/parser',
      },
    },
  ],
};
```
</details>

```svelte
<dialog bind:open={open}></dialog>
```

### What did you expect to happen?

Both [`<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#attributes) and [`<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#attributes) have an `open` attribute. Therefore, it should be supported in the latter.

### What actually happened?

> 'open' binding can only be used with `<details>`(invalid-binding)
>
> eslint([svelte/valid-compile](https://sveltejs.github.io/eslint-plugin-svelte/rules/valid-compile/))

### Link to **GitHub Repo** with Minimal Reproducible Example

Simply create a `<dialog>` element in a SvelteKit default template.

<img width="609" alt="image" src="https://github.com/sveltejs/eslint-plugin-svelte/assets/47051820/996218f8-5479-44d0-95e7-9ffe002676ba">

**Comment by hyunbinseo** [comment]
bind:open is not supported in Svelte.
- https://github.com/sveltejs/svelte/issues/4723
